src/libstore/build-result.cc

```diff
@@ -0,0 +1,18 @@
+#include "build-result.hh"
+
+namespace nix {
+
+GENERATE_CMP_EXT(
```

Comment:
Not something for this PR, but the documentation for this macro is quite awful. It took me a while to understand what it was doing (despite having more or less written the first version of it)